vllm/v1/engine/async_llm.py

```diff
@@ -249,19 +249,28 @@ async def _run_output_handler(self):
                 # Split outputs into chunks of at most
                 # VLLM_V1_OUTPUT_PROC_CHUNK_SIZE, so that we don't block the
                 # event loop for too long.
-                num_outputs = len(outputs.outputs)
-                if num_outputs <= VLLM_V1_OUTPUT_PROC_CHUNK_SIZE:
-                    slices = (outputs.outputs, )
+                num_outputs = len(outputs.new_token_id_offsets)
+
```

Comment:
You could do here `outputs.new_token_id_offsets.append(len(outputs.new_token_ids))` to avoid the special case logic inside `process_outputs()`.